document the _attrs_module classmethod (#6721)
sunpy/sunpy: docs/dev_guide/contents/extending_fido.rst

        return adict

Registering custom attrs in the ``attrs`` namespace
---------------------------------------------------

> When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.

Reviewer: ```suggestion

When you have custom attrs defined in a separate attrs module, you can add them to the main sunpy attrs namespace (``a.attrs``) using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.
```

> The method returns a tuple having the name of the target namespace and the module to be imported in that namespace.

Reviewer: ```suggestion
The method returns a tuple of length 2, where the first element is the name you want to add to the main attrs namespace. The second is module where the custom attrs exist, so that can be imported.
```

An example for this can be seen as implemented in the JSOC client:

.. code-block:: python

    @classmethod
    def _attrs_module(cls):
        return 'jsoc', 'sunpy.net.jsoc.attrs'

> 

Reviewer: ```suggestion

This adds the attrs that exist within ``sunpy.net.jsoc.attrs`` such as ``Keyword`` which can now be accessed by the man attrs import, e. g., ``a.jsoc.Keyword``.

This import can be any module, external as well.
```

        return adict

Registering custom attrs in the ``attrs`` namespace
---------------------------------------------------

> When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.

Reviewer: ```suggestion

When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.
```

When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.

> The method returns a tuple of length 2, where the first element is the target module name under which you want to add the custom attrs to the main attrs namespace. The second is the import path to the source module where the custom attrs are defined. Note that the source module here need not be an internal SunPy module, it could very well be external.

Reviewer: ```suggestion
The method returns a tuple of length 2, where the first element is the target module name under which you want to add the custom attrs to the main attrs namespace.
The second is the import path to the source module where the custom attrs are defined.
Note that the source module here need not be an internal ``sunpy`` module, it could very well be external.
```

        return adict

Registering custom attrs in the ``attrs`` namespace
---------------------------------------------------

> When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.

Reviewer: ```suggestion

When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.
```

Author: aaaah thanks very much 🙈 :skull_and_crossbones:

        return adict

Registering custom attrs in the ``attrs`` namespace
---------------------------------------------------

When you have custom attrs defined in a separate attrs module, you can add them to the namespace using the :meth:`~sunpy.net.BaseClient._attrs_module` class method.

> The method returns a tuple of length 2, where the first element is the name you want to add to the main attrs namespace. The second is module where the custom attrs exist, so that can be imported.

Reviewer 2: ```suggestion
The method returns a tuple of length 2, where the first element is the target module name under which you want to add the custom attrs to the main attrs namespace. The second is the import path to the source module where the custom attrs are defined.
```

> The method returns a tuple of length 2, where the first element is the target module name under which you want to add the custom attrs to the main attrs namespace. The second is the import path to the source module where the custom attrs are defined.

Author: ```suggestion
The method returns a tuple of length 2, where the first element is the target module name under which you want to add the custom attrs to the main attrs namespace. The second is the import path to the source module where the custom attrs are defined. Note that the source module here need not be an internal SunPy module, it could very well be external.
```

Author: does a small note here making the last line's point work? @alasdairwilson @nabobalis

Reviewer: Sure.

An example for this can be seen as implemented in the JSOC client:

.. code-block:: python

    @classmethod
    def _attrs_module(cls):
        return 'jsoc', 'sunpy.net.jsoc.attrs'

> This adds the attrs that exist within ``sunpy.net.jsoc.attrs`` such as ``Keyword`` which can now be accessed by the man attrs import, e. g., ``a.jsoc.Keyword``.

Reviewer 2: ```suggestion
This adds all attrs that exist within ``sunpy.net.jsoc.attrs``, such as ``Keyword``, to ``attrs.jsoc``. These can now be accessed via an import of the main attrs module, e. g. at ``a.jsoc.Keyword``.
```

> This import can be any module, external as well.

Reviewer 2: I am not clear on what this sentence means? Does it need clarifying (or removal)?

Reviewer: I was trying to see that this works for internal sunpy code, like the example but also for external libraries. Like DKIST.

Maybe we should remove it.

Author: does `The source module here need not be an internal SunPy module.` after the description of the second element of the tuple work?

Reviewer 2: Yeah I'm fine with the new line, remove it here and I'm happy.
